dodder-common/src/test/java/cc/dodder/common/entity/TreeTest.java

```diff
@@ -0,0 +1,107 @@
+  }
+
+  @Test
+  public void testConstructor() {
+    final Tree tree = new Tree("foo");
+    assertEquals(101605, tree.getRoot().hashCode());
+    assertEquals(-1, tree.getRoot().getIndex());
+    assertEquals(-1, tree.getRoot().getPid());
+    assertEquals(0, tree.getRoot().getNid());
+    assertEquals("foo", tree.getRoot().getFilename());
+    assertNull(((Node)tree.getRoot()).getChildren());
+    assertNull(tree.getRoot().getFilesize());
+  }
+
+  @Test
+  public void testCreateTree() {
+    final Tree tree = new Tree();
+    final ArrayList<Node> nodes = new ArrayList<>();
+    tree.setRoot(new Node());
+    nodes.add(new Node(0, 0, "foo", 0l, 0));
+    nodes.add(new Node(1, 1, "bar", 0l, 1));
+    tree.createTree(nodes);
+    assertEquals(new Node(0, 0, "foo", 0l, 0), tree.getRoot().getChildren().get(0));
+  }
+
+  @Test
+  public void testGetHtmlNoChildren() {
+    final Tree tree = new Tree();
+    tree.setRoot(new Node(0, 0, "a\'b\'c", 0l, 0));
+    assertEquals(
+            "<li><span class=\"file\">a'b'c<small>(0 B)</small></span></li>",
+            tree.getHtml(new Node(0, 0, "a\'b\'c", 0l, 0)));
+  }
+
+  @Test
+  public void testGetHtmlMatchingPid() {
+    final Tree tree = new Tree();
+    tree.setRoot(new Node(0, 0, "a\'b\'c", 0l, 0));
+    final Node node = new Node(0, 0, "a/b/c", 0l, 0);
+    node.setChildren(new ArrayList<Node>());
+    assertEquals(
+            "<ul class=\"filetree treeview\"><p><span class=\"bticon\">a\'b\'c</span></p></ul></li>",
+            tree.getHtml(node));
+  }
+
+  @Test
+  public void testGetHtmlNotMatchingPid() {
+    final Tree tree = new Tree();
+    tree.setRoot(new Node(0, 0, "a\'b\'c", 0l, 0));
+    final Node node = new Node(-2_147_483_648, 0, "a/b/c", 0l, 0);
+    node.setChildren(new ArrayList<Node>());
+    assertEquals("<li class=\"closed\"><span class=\"folder\">a/b/c</span><ul></ul></li>",
+            tree.getHtml(node));
+  }
+
+  @Test
+  public void testGetLeafList() {
+    final Tree tree = new Tree();
+    tree.setRoot(new Node(0, 0, "foo", 0l, 0));
+    final List<Node> nodes = tree.getLeafList();
+    assertEquals(1, nodes.size());
+    assertEquals(0, nodes.get(0).getIndex());
+    assertEquals(0, nodes.get(0).getPid());
+    assertEquals(0, nodes.get(0).getNid());
+    assertEquals("foo", nodes.get(0).getFilename());
+  }
+
+  @Test
+  public void testGetLeafListChildrenNotNull() {
+    final Tree tree = new Tree();
+    final Node node = new Node();
+    node.setChildren(new ArrayList());
+    tree.setRoot(node);
+    assertEquals(new ArrayList<Node>(), tree.getLeafList());
+  }
+
+  @Test
+  public void testMiddlePrint() {
+    final Tree objectUnderTest = new Tree();
+    objectUnderTest.setRoot(null);
+    final Node node = new Node();
+    final ArrayList<Node> arrayList = new ArrayList<Node>();
+    arrayList.add(new Node());
+    node.setChildren(arrayList);
+    objectUnderTest.middlePrint(node);
+
+    // Method returns void, testing that no exception is thrown
```

Comment:
Are there no side effects from this method that you can assert instead?

Reply:
The method just recursively prints names to the console so unless I should be trying to read this there are no there side effects

Reply:
@ThomasPerkins1123 https://stackoverflow.com/questions/1119385/junit-test-for-system-out-println Maybe this might work? Assuming you can determine how many recursions it will take?